src/bentoml/_internal/service/openapi/utils.py

```diff
@@ -43,6 +75,33 @@ def pydantic_components_schema(pydantic_model: t.Type[pydantic.BaseModel]):
     return {k: Schema(**definitions[k]) for k in sorted(definitions)}
 
 
+def typeddict_components_schema(typeddict: t.Type[TypedDict]):
```

Comment:
```suggestion
def typeddict_components_schema(typeddict: type[TypedDict]):
```

Reply:
I'm also ok with this section, but it seems to me that this section is something that pydantic already do.

Maybe a way to reduce coverage for this logic is to convert the user typeddict to pydantic model, and then use the logic we already have for converting pydantic model to the schema

Reply:
I think that it is good way to implement `typeddict_components_schema()`  
because i'm concerned about potential bugs during maintenance that will arise from implementing `typeddict_components_schema()` bentoml itself not depend on pydantic.

but it is awkward to bentoml user, when they use typeddict, should be need extra install `bentoml[pydantic]`
`nitpick:`  I think It would be nice to have pydantic as a default dependency module in bentoml instead of an extra dependency module.

IMO these is no good way or bad way, 
it just would be trade off, depend on pydantic or not 
for now, i will make an additional commit using pydantic. :)

Reply:
Now I have more time to think about this, if we want to support TypedDict first class, then we shouldn't use pydantic

Reply:
We probably dont' want to depend on pydantic.